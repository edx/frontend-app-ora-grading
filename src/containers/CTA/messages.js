/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  ctaFeedbackMessage: {
    id: 'ora-grading.CTA.feedbackMessage',
    defaultMessage: 'Thanks for using the new ORA staff grading experience. We\'d love to hear your feedback',
    description: 'Thank user for using ora and ask for feed back',
  },
  ctaLinkMessage: {
    id: 'ora-grading.CTA.linkMessage',
    defaultMessage: 'here',
    description: 'placeholder for the feedback anchor link',
  },
});

export default StrictDict(messages);
