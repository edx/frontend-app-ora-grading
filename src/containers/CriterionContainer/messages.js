import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  addComments: {
    id: 'ora-grading.CriterionFeedback.addCommentsLabel',
    defaultMessage: 'Add comments',
    description: 'label for editable feedback field',
  },
  comments: {
    id: 'ora-grading.CriterionFeedback.commentsLabel',
    defaultMessage: 'Comments',
    description: 'label for read-only feedback field',
  },
  optionPoints: {
    id: 'ora-grading.RadioCriterion.optionPoints',
    defaultMessage: '{points} points',
    description: 'criterion option point value display',
  },
});

export default messages;
