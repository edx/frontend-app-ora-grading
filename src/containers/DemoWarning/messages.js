/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  demoModeHeading: {
    id: 'ora-grading.ReviewModal.demoHeading',
    defaultMessage: 'Demo Mode',
    description: 'Demo mode heading',
  },
  demoModeMessage: {
    id: 'ora-grading.ReviewModal.demoMessage',
    defaultMessage: 'You are using the Demo Mode of the new Enhanced ORA Staff Grader interface.  You will be unable to submit grades until you activate the feature.',
    description: 'Demo mode message',
  },
});

export default StrictDict(messages);
