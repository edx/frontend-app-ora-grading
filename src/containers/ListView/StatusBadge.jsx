import React from 'react';
import PropTypes from 'prop-types';

import { Badge } from '@edx/paragon';

import { gradingStatuses as statuses } from 'data/services/lms/constants';

/**
 * <StatusBadge />
 */
export const StatusBadge = ({ status }) => {
  if (status === statuses.ungraded) {
    return (<Badge variant="primary">Ungraded</Badge>);
  }
  if (status === statuses.locked) {
    return (<Badge variant="light">Grading in progress</Badge>);
  }
  if (status === statuses.graded) {
    return (<Badge variant="success">Grading Complete</Badge>);
  }
  return (<Badge>{status}</Badge>);
};
StatusBadge.propTypes = {
  status: PropTypes.string.isRequired,
};

export default StatusBadge;
