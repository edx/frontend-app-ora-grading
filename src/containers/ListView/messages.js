import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  backToResponses: {
    id: 'ora-grading.ListView.ListViewBreadcrumbs.backToResponses',
    defaultMessage: 'Back to all open responses',
    description: 'Breadcrumbs link text to return to ORA list in LMS.',
  },
  noResultsFound: {
    id: 'ora-grading.ListView.noResultsFound',
    defaultMessage: 'No results found',
    description: 'Empty table content for submissions list',
  },
  viewAllResponses: {
    id: 'ora-grading.ListView.viewAllResponses',
    defaultMessage: 'View all responses',
    description: 'Button text to load all responses for review/grading',
  },
  viewSelectedResponses: {
    id: 'ora-grading.ListView.viewSelectedResponses',
    defaultMessage: 'View selected responses ({value})',
    description: 'Button text to load selected responses for review/grading',
  },
  username: {
    id: 'ora-grading.ListView.tableHeaders.username',
    defaultMessage: 'Username',
    description: 'Username table column header for submission list view',
  },
  teamName: {
    id: 'ora-grading.ListView.tableHeaders.teamName',
    defaultMessage: 'Team name',
    description: 'Team name table column header for submission list view',
  },
  learnerSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.learnerSubmissionDate',
    defaultMessage: 'Learner submission date',
    description: 'Learner submission date table column header for submission list view',
  },
  teamSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.teamSubmissionDate',
    defaultMessage: 'Team submission date',
    description: 'Team submission date table column header for submission list view',
  },
  grade: {
    id: 'ora-grading.ListView.tableHeaders.grade',
    defaultMessage: 'Grade',
    description: 'Grade table column header for submission list view',
  },
  gradingStatus: {
    id: 'ora-grading.ListView.tableHeaders.gradingStatus',
    defaultMessage: 'Grading status',
    description: 'Grading status table column header for submission list view',
  },
});

export default messages;
