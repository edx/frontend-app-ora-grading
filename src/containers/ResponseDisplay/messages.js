import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  tableNameHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableNameHeader',
    defaultMessage: 'Name',
    description: 'Table header file name',
  },
  tableExtensionHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableExtensionHeader',
    defaultMessage: 'File Extension',
    description: 'Table header file extension',
  },
  tablePopoverHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tablePopoverHeader',
    defaultMessage: 'File Metadata',
    description: 'Table header for popover file metadata',
  },
});

export default messages;
