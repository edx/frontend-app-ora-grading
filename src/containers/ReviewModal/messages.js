import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  reloadSubmission: {
    id: 'ora-grading.ReviewModal.reloadSubmission',
    defaultMessage: 'Reload submission',
    description: 'Reload button text in case of network failure',
  },
  loadingResponse: {
    id: 'ora-grading.ReviewModal.loadingResponse',
    defaultMessage: 'Loading response',
    description: 'loading text for submission response review screen',
  },
  gradeNotSubmittedHeading: {
    id: 'ora-grading.ReviewModal.gradeNotSubmitted.heading',
    defaultMessage: 'Grade not submitted',
    description: 'Grade submission network error heading',
  },
  gradeNotSubmittedContent: {
    id: 'ora-grading.ReviewModal.gradeNotSubmitted.heading',
    defaultMessage: "We're sorry, something went wrong when we tried to submit this grade.  Please try again.",
    description: 'Grade submission network error message',
  },
  resubmitGrade: {
    id: 'ora-grading.ReviewModal.resubmitGrade',
    defaultMessage: 'Resubmit grate',
    description: 'Resubmit grade button after network failure',
  },
  dismiss: {
    id: 'ora-grading.ReviewModal.dismiss',
    defaultMessage: 'Dismiss',
    description: 'Dismiss error action button text',
  },
});

export default StrictDict(messages);
