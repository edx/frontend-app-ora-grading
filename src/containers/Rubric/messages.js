import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  rubric: {
    id: 'ora-grading.Rubric.rubric',
    defaultMessage: 'Rubric',
    description: 'Rubric interface label',
  },
  submitGrade: {
    id: 'ora-grading.Rubric.submitGrade',
    defaultMessage: 'Submit grade',
    description: 'Submit Grade button text',
  },
  overallComments: {
    id: 'ora-grading.Rubric.overallComments',
    defaultMessage: 'Overall comments',
    description: 'Rubric overall commnents label',
  },
  addComments: {
    id: 'ora-grading.Rubric.addComments',
    defaultMessage: 'Add comments',
    description: 'Rubric comments input label',
  },
  comments: {
    id: 'ora-grading.Rubric.comments',
    defaultMessage: 'Comments',
    description: 'Rubric comments display label',
  },
});

export default messages;
