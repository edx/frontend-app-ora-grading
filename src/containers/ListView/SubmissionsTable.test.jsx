import React from 'react';
import { shallow } from 'enzyme';

import {
  DataTable,
  MultiSelectDropdownFilter,
  TextFilter,
} from '@edx/paragon';

import { selectors, thunkActions } from 'data/redux';
import { gradingStatuses as statuses, submissionFields } from 'data/services/lms/constants';

import StatusBadge from 'components/StatusBadge';
import { formatMessage } from 'testUtils';
import messages from './messages';
import {
  SubmissionsTable,
  mapStateToProps,
  mapDispatchToProps,
} from './SubmissionsTable';

jest.mock('./FilterStatusComponent', () => jest.fn().mockName('FilterStatusComponent'));
jest.mock('./TableAction', () => jest.fn().mockName('TableAction'));
jest.mock('./SelectedBulkAction', () => jest.fn().mockName('SelectedBulkAction'));

jest.mock('data/redux', () => ({
  selectors: {
    app: {
      ora: {
        isIndividual: (...args) => ({ isIndividual: args }),
      },
    },
    submissions: {
      listData: (...args) => ({ listData: args }),
    },
  },
  thunkActions: {
    grading: {
      loadSelectionForReview: (...args) => ({ loadSelectionForReview: args }),
    },
  },
}));

let el;
jest.useFakeTimers('modern');

const dates = [
  '2021-12-08 09:06:15.319213+00:00',
  '2021-12-10 18:06:15.319213+00:00',
  '2021-12-11 07:06:15.319213+00:00',
];

const individualData = [
  {
    username: 'username-1',
    dateSubmitted: dates[0],
    gradingStatus: statuses.ungraded,
    score: {
      pointsEarned: 1,
      pointsPossible: 10,
    },
  },
  {
    username: 'username-2',
    dateSubmitted: dates[1],
    gradingStatus: statuses.graded,
    score: {
      pointsEarned: 2,
      pointsPossible: 10,
    },
  },
  {
    username: 'username-3',
    dateSubmitted: dates[2],
    gradingStatus: statuses.inProgress,
    score: {
      pointsEarned: 3,
      pointsPossible: 10,
    },
  },
];

const teamData = [
  {
    teamName: 'teamName-1',
    dateSubmitted: dates[0],
    gradingStatus: statuses.ungraded,
    score: {
      pointsEarned: 1,
      pointsPossible: 10,
    },
  },
  {
    teamName: 'teamName-2',
    dateSubmitted: dates[1],
    gradingStatus: statuses.graded,
    score: {
      pointsEarned: 2,
      pointsPossible: 10,
    },
  },
  {
    teamName: 'teamName-3',
    dateSubmitted: dates[2],
    gradingStatus: statuses.inProgress,
    score: {
      pointsEarned: 3,
      pointsPossible: 10,
    },
  },
];

describe('SubmissionsTable component', () => {
  describe('component', () => {
    const props = {
      isIndividual: true,
      listData: [...individualData],
    };
    beforeEach(() => {
      props.loadSelectionForReview = jest.fn();
      props.intl = { formatMessage };
    });
    describe('render tests', () => {
      const mockMethod = (methodName) => {
        el.instance()[methodName] = jest.fn().mockName(`this.${methodName}`);
      };
      beforeEach(() => {
        el = shallow(<SubmissionsTable {...props} />);
      });
      describe('snapshots', () => {
        beforeEach(() => {
          mockMethod('handleViewAllResponsesClick');
          mockMethod('formatDate');
          mockMethod('formatGrade');
          mockMethod('formatStatus');
        });
        test('snapshot: empty (no list data)', () => {
          el = shallow(<SubmissionsTable {...props} listData={[]} />);
          expect(el).toMatchSnapshot();
          expect(el.isEmptyRender()).toEqual(true);
        });
        test('snapshot: happy path', () => {
          expect(el.instance().render()).toMatchSnapshot();
        });
        test('snapshot: team happy path', () => {
          el.setProps({ isIndividual: false, listData: [...teamData] });
          expect(el.instance().render()).toMatchSnapshot();
        });
      });
      describe('DataTable', () => {
        let table;
        let tableProps;
        beforeEach(() => {
          table = el.find(DataTable);
          tableProps = table.props();
        });
        test.each([
          'isFilterable',
          'isSelectable',
          'isSortable',
          'isPaginated',
        ])('%s', key => expect(tableProps[key]).toEqual(true));
        test.each([
          ['numBreakoutFilters', 2],
          ['defaultColumnValues', { Filter: TextFilter }],
          ['itemCount', 3],
          ['initialState', { pageSize: 10, pageIndex: 0 }],
        ])('%s = %p', (key, value) => expect(tableProps[key]).toEqual(value));
        describe('individual columns', () => {
          let columns;
          beforeEach(() => {
            columns = tableProps.columns;
          });
          test('username column', () => {
            expect(columns[0]).toEqual({
              Header: messages.username.defaultMessage,
              accessor: submissionFields.username,
            });
          });
          test('submission date column', () => {
            expect(columns[1]).toEqual({
              Header: messages.learnerSubmissionDate.defaultMessage,
              accessor: submissionFields.dateSubmitted,
              Cell: el.instance().formatDate,
              disableFilters: true,
            });
          });
          test('grade column', () => {
            expect(columns[2]).toEqual({
              Header: messages.grade.defaultMessage,
              accessor: submissionFields.score,
              Cell: el.instance().formatGrade,
              disableFilters: true,
            });
          });
          test('grading status column', () => {
            expect(columns[3]).toEqual({
              Header: messages.gradingStatus.defaultMessage,
              accessor: submissionFields.gradingStatus,
              Cell: el.instance().formatStatus,
              Filter: MultiSelectDropdownFilter,
              filter: 'includesValue',
              filterChoices: el.instance().gradeStatusOptions,
            });
          });
        });
        describe('team columns', () => {
          let columns;
          beforeEach(() => {
            el.setProps({ isIndividual: false, listData: [...teamData] });
            columns = el.find(DataTable).props().columns;
          });
          test('teamName column', () => {
            expect(columns[0]).toEqual({
              Header: messages.teamName.defaultMessage,
              accessor: submissionFields.teamName,
            });
          });
          test('submission date column', () => {
            expect(columns[1]).toEqual({
              Header: messages.teamSubmissionDate.defaultMessage,
              accessor: submissionFields.dateSubmitted,
              Cell: el.instance().formatDate,
              disableFilters: true,
            });
          });
          test('grade column', () => {
            expect(columns[2]).toEqual({
              Header: messages.grade.defaultMessage,
              accessor: submissionFields.score,
              Cell: el.instance().formatGrade,
              disableFilters: true,
            });
          });
          test('grading status column', () => {
            expect(columns[3]).toEqual({
              Header: messages.gradingStatus.defaultMessage,
              accessor: submissionFields.gradingStatus,
              Cell: el.instance().formatStatus,
              Filter: MultiSelectDropdownFilter,
              filter: 'includesValue',
              filterChoices: el.instance().gradeStatusOptions,
            });
          });
        });
      });
    });
    describe('behavior', () => {
      describe('formatDate method', () => {
        it('returns the date in locale time string', () => {
          const fakeDate = 16131215154955;
          const fakeDateString = 'test-date-string';
          const mock = jest.spyOn(Date.prototype, 'toLocaleString').mockReturnValue(fakeDateString);
          expect(el.instance().formatDate({ value: fakeDate })).toEqual(fakeDateString);
          mock.mockRestore();
        });
      });
      describe('formatGrade method', () => {
        it('returns "-" if grade is null', () => {
          expect(el.instance().formatGrade({ value: null })).toEqual('-');
        });
        it('returns <pointsEarned>/<pointsPossible> if grade exists', () => {
          expect(
            el.instance().formatGrade({ value: { pointsEarned: 1, pointsPossible: 10 } }),
          ).toEqual('1/10');
        });
      });
      describe('formatStatus method', () => {
        it('returns a StatusBadge with the given status', () => {
          const status = 'graded';
          expect(el.instance().formatStatus({ value: 'graded' })).toEqual(
            <StatusBadge status={status} />,
          );
        });
      });
      describe('handleViewAllResponsesClick', () => {
        it('calls loadSelectionForReview with submissionUUID from all rows if there are no selectedRows', () => {
          const data = [
            { original: { submissionUUID: '123' } },
            { original: { submissionUUID: '456' } },
            { original: { submissionUUID: '789' } },
          ];
          el.instance().handleViewAllResponsesClick(data)();
          expect(el.instance().props.loadSelectionForReview).toHaveBeenCalledWith(['123', '456', '789']);
        });
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('listData loads from submissions.listData', () => {
      expect(mapped.listData).toEqual(selectors.submissions.listData(testState));
    });
  });
  describe('mapDispatchToProps', () => {
    it('loads loadSelectionForReview from thunkActions.grading.loadSelectionForReview', () => {
      expect(
        mapDispatchToProps.loadSelectionForReview,
      ).toEqual(thunkActions.grading.loadSelectionForReview);
    });
  });
});
