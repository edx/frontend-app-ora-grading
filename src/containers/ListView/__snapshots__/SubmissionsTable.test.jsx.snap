// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionsTable component component render tests snapshots snapshot: empty (no list data) 1`] = `""`;

exports[`SubmissionsTable component component render tests snapshots snapshot: happy path 1`] = `
<DataTable
  FilterStatusComponent={[MockFunction FilterStatusComponent]}
  bulkActions={
    Array [
      [MockFunction this.selectedBulkAction],
    ]
  }
  columns={
    Array [
      Object {
        "Header": "Username",
        "accessor": "username",
      },
      Object {
        "Cell": [MockFunction this.formatDate],
        "Header": "Learner submission date",
        "accessor": "dateSubmitted",
        "disableFilters": true,
      },
      Object {
        "Cell": [MockFunction this.formatGrade],
        "Header": "Grade",
        "accessor": "score",
        "disableFilters": true,
      },
      Object {
        "Cell": [MockFunction this.formatStatus],
        "Filter": "MultiSelectDropdownFilter",
        "Header": "Grading status",
        "accessor": "gradingStatus",
        "filter": "includesValue",
        "filterChoices": Array [
          Object {
            "name": "Ungraded",
            "value": "ungraded",
          },
          Object {
            "name": "Grading Completed",
            "value": "graded",
          },
          Object {
            "name": "Currently being graded by someone else",
            "value": "locked",
          },
          Object {
            "name": "You are currently grading this response",
            "value": "in-progress",
          },
        ],
      },
    ]
  }
  data={
    Array [
      Object {
        "dateSubmitted": "9:05:54 PM",
        "gradingStatus": "ungraded",
        "score": Object {
          "pointsEarned": 1,
          "pointsPossible": 10,
        },
        "username": "username-1",
      },
      Object {
        "dateSubmitted": "11:52:34 PM",
        "gradingStatus": "graded",
        "score": Object {
          "pointsEarned": 2,
          "pointsPossible": 10,
        },
        "username": "username-2",
      },
      Object {
        "dateSubmitted": "9:07:30 PM",
        "gradingStatus": "in-progress",
        "score": Object {
          "pointsEarned": 3,
          "pointsPossible": 10,
        },
        "username": "username-3",
      },
    ]
  }
  defaultColumnValues={
    Object {
      "Filter": "TextFilter",
    }
  }
  initialState={
    Object {
      "pageIndex": 0,
      "pageSize": 10,
    }
  }
  isFilterable={true}
  isPaginated={true}
  isSelectable={true}
  isSortable={true}
  itemCount={3}
  numBreakoutFilters={2}
  tableActions={
    Array [
      Object {
        "buttonText": "View all responses",
        "className": "view-all-responses-btn",
        "handleClick": [MockFunction this.handleViewAllResponsesClick],
        "variant": "primary",
      },
    ]
  }
>
  <DataTable.TableControlBar />
  <DataTable.Table />
  <DataTable.TableFooter />
</DataTable>
`;

exports[`SubmissionsTable component component render tests snapshots snapshot: team happy path 1`] = `
<DataTable
  FilterStatusComponent={[MockFunction FilterStatusComponent]}
  bulkActions={
    Array [
      [MockFunction this.selectedBulkAction],
    ]
  }
  columns={
    Array [
      Object {
        "Header": "Team name",
        "accessor": "teamName",
      },
      Object {
        "Cell": [MockFunction this.formatDate],
        "Header": "Team submission date",
        "accessor": "dateSubmitted",
        "disableFilters": true,
      },
      Object {
        "Cell": [MockFunction this.formatGrade],
        "Header": "Grade",
        "accessor": "score",
        "disableFilters": true,
      },
      Object {
        "Cell": [MockFunction this.formatStatus],
        "Filter": "MultiSelectDropdownFilter",
        "Header": "Grading status",
        "accessor": "gradingStatus",
        "filter": "includesValue",
        "filterChoices": Array [
          Object {
            "name": "Ungraded",
            "value": "ungraded",
          },
          Object {
            "name": "Grading Completed",
            "value": "graded",
          },
          Object {
            "name": "Currently being graded by someone else",
            "value": "locked",
          },
          Object {
            "name": "You are currently grading this response",
            "value": "in-progress",
          },
        ],
      },
    ]
  }
  data={
    Array [
      Object {
        "dateSubmitted": "9:05:54 PM",
        "gradingStatus": "ungraded",
        "score": Object {
          "pointsEarned": 1,
          "pointsPossible": 10,
        },
        "teamName": "teamName-1",
      },
      Object {
        "dateSubmitted": "11:52:34 PM",
        "gradingStatus": "graded",
        "score": Object {
          "pointsEarned": 2,
          "pointsPossible": 10,
        },
        "teamName": "teamName-2",
      },
      Object {
        "dateSubmitted": "9:07:30 PM",
        "gradingStatus": "in-progress",
        "score": Object {
          "pointsEarned": 3,
          "pointsPossible": 10,
        },
        "teamName": "teamName-3",
      },
    ]
  }
  defaultColumnValues={
    Object {
      "Filter": "TextFilter",
    }
  }
  initialState={
    Object {
      "pageIndex": 0,
      "pageSize": 10,
    }
  }
  isFilterable={true}
  isPaginated={true}
  isSelectable={true}
  isSortable={true}
  itemCount={3}
  numBreakoutFilters={2}
  tableActions={
    Array [
      Object {
        "buttonText": "View all responses",
        "className": "view-all-responses-btn",
        "handleClick": [MockFunction this.handleViewAllResponsesClick],
        "variant": "primary",
      },
    ]
  }
>
  <DataTable.TableControlBar />
  <DataTable.Table />
  <DataTable.TableFooter />
</DataTable>
`;
